.github/workflows/build_test_clippy.yml

```diff
@@ -0,0 +1,27 @@
+name: Build, test and Clippy
+
+on:
+  push:
+  pull_request:
+    branches: [ "main" ]
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  build-test-clippy:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+    - uses: dtolnay/rust-toolchain@master
+      with:
+        toolchain: nightly-2022-11-21
+        components: clippy
+    - name: Build
+      run: cargo build --verbose
+    - name: Run tests
+      run: cargo test --verbose
+    - name: Run clippy
+      run: cargo clippy --verbose
```

Comment:
Has this changed vs what's in main?

Reply:
I merged main into this. Yes, still the same.